mysql-test/suite/json/t/json_no_table.test

```diff
@@ -1484,6 +1484,46 @@ select json_array(b'0', b'1', b'10');
 # returns the empty array: []
 SELECT JSON_ARRAY();
 
+
+--echo # ----------------------------------------------------------------------
+--echo # Bug #MDEV-23143: missing a JSON_EQUALS function
+--echo # ----------------------------------------------------------------------
+
+--echo # should give NULL
+select JSON_EQUALS(NULL, NULL);
+select JSON_EQUALS(json_compact('{"a": 1, "b": 2}'), NULL);
+select JSON_EQUALS(NULL, '[1, 2]');
+
+--echo # should give 0: no type conversion
+select JSON_EQUALS(json_compact(3.14), json_compact(3));
+
+--echo # returns 0: one candidate element doesn't match
+SELECT JSON_EQUALS('{"A": 0, "B": {"C": 1}, "D": 2}', '{"A": 7, "C": 9, "B": 8}');
+SELECT JSON_EQUALS('{"A": 0, "B": {"C": 1}, "D": 2}', '{"A": 0, "B": {"C": 2}, "D": 2}');
+
+--echo #returns 1
+SELECT JSON_EQUALS('{"A": 0, "B": {"C": 1}, "D": 2}', '{"A": 0, "B": {"C": 1}, "D": 2}');
```

Comment:
What about a different key order in either JSON document?

`{"A":0, "B":1}` === `{"B":1, "A":0}`